lib/ansible/plugins/vars/host_group_vars.py

```diff
@@ -128,7 +134,7 @@ def _get_dir_files(self, path):
 
                 if os.path.isdir(full_spath) and not ext:  # recursive search if dir
                     found.extend(self._get_dir_files(full_spath))
-                elif os.path.isfile(full_spath) and (not ext or ext in C.YAML_FILENAME_EXTENSIONS):
+                elif os.path.isfile(full_spath) and (not ext or ext in C.YAML_FILENAME_EXTENSIONS) and not self._is_ext_blacklisted(full_spath):
```

Comment:
true if var file has extension, but it is allowed to have var files with no extension in var subdirectories which results in something e.g. group_vars/all/file~